unittest/client/src/ilm_control_unittests.cpp

```diff
@@ -0,0 +1,5071 @@
+ *                         +# ilm_takeScreenshot() time 2 should return ILM_FAILED
+ */
+TEST_F(IlmControlTest, ilm_takeScreenshot_validScreen)
+{
+    t_ilm_uint lScreenID{10};
+    t_ilm_const_string filename(NULL);
+    /*- valid screen ID
+      - invalid file name*/
+    ASSERT_EQ(ILM_SUCCESS, ilm_takeScreenshot(lScreenID, filename));
+    /*- valid screen ID
+      - valid file name*/
+    filename = "name_test";
+    ASSERT_EQ(ILM_FAILED, ilm_takeScreenshot(lScreenID, filename));
+}
+
+/** ================================================================================================
+ * @test_id             ilm_takeScreenshot_SupportArgb8888AndUnsupportWlshmGlobal
+ * @brief               ilm_takeScreenshot() is called when argb8888 is supported and wl_shm global is unsupported
+ * @test_procedure Steps:
+ *                      -# make argb8888 is supported
+ *                      -# point wl_shm to NULL
+ *                      -# call ilm_takeScreenshot with valid screenID and file name
+ *                      -# Verification point:
+ *                         +# ilm_takeScreenshot() must return ILM_FAILED
+ *                         +# wl_proxy_marshal_flags should not be called
+ */
+TEST_F(IlmControlTest, ilm_takeScreenshot_SupportArgb8888AndUnsupportWlshmGlobal)
+{
+    t_ilm_uint lScreenID{10};
+    t_ilm_const_string filename{"name_test"};
+    constexpr uint32_t lValidAddress{0xFFFFFFFF};
+
+    ilm_context.wl.wl_shm = NULL;
+    ilm_context.wl.has_argb8888 = true;
+    ASSERT_EQ(ILM_FAILED, ilm_takeScreenshot(lScreenID, filename));
+    ASSERT_EQ(wl_proxy_marshal_flags_fake.call_count, 0);
+    ilm_context.wl.wl_shm = NULL;
+    ilm_context.wl.has_argb8888 = false;
+}
+
+/** ================================================================================================
+ * @test_id             ilm_takeScreenshot_SupportWlshmGlobalAndUnsupportArgb8888
+ * @brief               ilm_takeScreenshot() is called when wl_shm global is supported and argb8888 is unsupported
+ * @test_procedure Steps:
+ *                      -# make wl_shm global is supported
+ *                      -# make argb8888 is unsupported
+ *                      -# call ilm_takeScreenshot with valid screenID and file name
+ *                      -# Verification point:
+ *                         +# ilm_takeScreenshot() must return ILM_FAILED
+ *                         +# wl_proxy_marshal_flags should not be called
+ */
+TEST_F(IlmControlTest, ilm_takeScreenshot_SupportWlshmGlobalAndUnsupportArgb8888)
+{
+    t_ilm_uint lScreenID{10};
+    t_ilm_const_string filename{"name_test"};
+    constexpr uint32_t lValidAddress{0xFFFFFFFF};
+    // prepare data
+    ilm_context.wl.wl_shm = lValidAddress;
+    ilm_context.wl.has_argb8888 = false;
+    // call and verify data
+    ASSERT_EQ(ILM_FAILED, ilm_takeScreenshot(lScreenID, filename));
+    ASSERT_EQ(wl_proxy_marshal_flags_fake.call_count, 0);
+    // revert to default data
+    ilm_context.wl.wl_shm = NULL;
```

Comment:
This should be taken care in either TearDown or SetUp need not be separately handled inside test case